akka-http-tests/src/test/scala/akka/http/scaladsl/marshalling/ContentNegotiationSpec.scala

```diff
@@ -136,6 +136,40 @@ class ContentNegotiationSpec extends FreeSpec with Matchers {
       accept(`text/plain`, `text/xml`) should select(`text/plain` withCharset `UTF-8`)
       accept(`text/xml`, `text/plain`) should select(`text/xml` withCharset `UTF-8`)
     }
+
+    /** when both acceptable media types are equally specific and have the same qValue, we choose to send the first */
+    "Accept: text/html;level=1, text/html;format=flowed" test { accept ⇒
+      accept(`text/html` withParams Map("level" → "1"), `text/html` withParams Map("format" → "flowed")) should select(`text/html` withParams Map("level" → "1") withCharset `UTF-8`)
+    }
+
+    /** when an acceptable media type is more specific, it is matched even when it has a lower qValue */
+    "Accept: text/html;format=flowed;q=0.5, text/html;q=1" test { accept ⇒
+      accept(`text/html` withParams Map("level" → "1"), `text/html` withParams Map("format" → "flowed")) should select(`text/html` withParams Map("level" → "1") withCharset `UTF-8`)
+    }
+
+    /** when both acceptable media types are available as separate marshallers and equally specific, we must select the one with the highest qValue */
```

Comment:
Ah I agree this comment is confusing, 'equally specific' doesn't have anything to do with the choice here because each marshaller only matches 1 of the types from the Accept header, so a better comment would be simply "when both acceptable media types are available as separate marshallers, we must select the one with the highest qValue"